gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -0,0 +1,169 @@
+import org.apache.gobblin.service.modules.orchestration.DagManager;
+
+
+/**
+ * A DagActionStore change monitor that uses {@link DagActionStoreChangeEvent} schema to process Kafka messages received
+ * from its corresponding consumer client. This monitor responds to requests to resume or delete a flow and acts as a
+ * connector between the API and execution layers of GaaS.
+ */
+@Slf4j
+public class DagActionStoreChangeMonitor extends HighLevelConsumer {
+  public static final String DAG_ACTION_CHANGE_MONITOR_PREFIX = "dagActionChangeStore";
+
+  // Metrics
+  ContextAwareMeter killsInvoked;
+  ContextAwareMeter resumesInvoked;
+  ContextAwareMeter unexpectedErrors;
+
+  protected CacheLoader<String, String> cacheLoader = new CacheLoader<String, String>() {
+    @Override
+    public String load(String key) throws Exception {
+      return key;
+    }
+  };
+
+  protected LoadingCache<String, String>
+      dagActionsSeenCache = CacheBuilder.newBuilder().expireAfterWrite(10, TimeUnit.MINUTES).build(cacheLoader);
+
+  @Inject
+  protected DagActionStore dagActionStore;
+
+  @Inject
+  protected DagManager dagManager;
+
+  public DagActionStoreChangeMonitor(String topic, Config config, int numThreads) {
+    // Differentiate group id for each host
+    super(topic, config.withValue(GROUP_ID_KEY,
+        ConfigValueFactory.fromAnyRef(DAG_ACTION_CHANGE_MONITOR_PREFIX + UUID.randomUUID().toString())),
+        numThreads);
+  }
+
+  @Override
+  protected void assignTopicPartitions() {
+    ChangeMonitorUtils.assignTopicPartitionsHelper(this.topic, this.getGobblinKafkaConsumerClient());
+  }
+
+  @Override
+  /*
+  This class is multi-threaded and this message will be called by multiple threads, however any given message will be
+  partitioned and processed by only one thread (and corresponding queue).
+   */
+  protected void processMessage(DecodeableKafkaRecord message) {
+    String key = (String) message.getKey();
+    DagActionStoreChangeEvent value = (DagActionStoreChangeEvent) message.getValue();
+
+    Long timestamp = value.getChangeEventIdentifier().getTimestamp();
+    String operation = value.getChangeEventIdentifier().getOperationType().name();
+    String flowGroup = value.getFlowGroup();
+    String flowName = value.getFlowName();
+    String flowExecutionId = value.getFlowExecutionId();
+
+    log.debug("Processing Dag Action message for flow group: {} name: {} executionId: {} timestamp {} operation {}",
+        flowGroup, flowName, flowExecutionId, timestamp, operation);
+
+    String changeIdentifier = timestamp + key;
+    if (!ChangeMonitorUtils.shouldProcessMessage(changeIdentifier, dagActionsSeenCache, operation,
+        timestamp.toString())) {
+      return;
+    }
+
+    // retrieve operation type from MySQL table OR from the event itself
+    DagActionStore.DagActionValue dagAction = null;
+    try {
+      dagAction = dagActionStore.getDagAction(flowGroup, flowName, flowExecutionId).getDagActionValue();
```

Comment:
Again, if the operation is delete, you don't need to do anything here right? why we want get dag even in that case?

Reply:
i'll retrieve only if operation is not delete